### PR TITLE
Improve logging macros to work with IDEs better

### PR DESCRIPTION
**Pull request by lrubasze**
## Summary
Improve logging macros to work with IDEs better

## Testing
Respective tests added

## Update Recommendations
Change is backward-compatible. No actions required

### CODE REVIEW

File: scrypto/src/macros.rs [hunk 1]
```diff
@@ -17,7 +17,16 @@ macro_rules! error {
...
 #[macro_export]
 macro_rules! error {
-    ($($args: expr),+) => {{}};
+    ($($args: expr),+) => {{
+        // The operation below does nothing at runtime but consumes the expressions,
+        // allowing tools (e.g., rust-analyzer) to parse them correctly to identify
+        // syntactical errors or provide hints and completions.
+        // Using `false` ensures the expression is optimized out in case of expressions
+        // that mutate data or have side effects.
+        if false {
```
Comment (dhedey) [inline scrypto/src/macros.rs:26]: I'm surprised this doesn't give an unreachable code lint. I wouldn't be surprised if such a lint were added some time. Perhaps we should make it this:
```rust
#[allow(unreachable_code)]
if false {
    let _ = ::scrypto::prelude::sbor::rust::format!($($args),+);
}
```

**Review by dhedey — APPROVED** [review]
Nice!

File: scrypto/src/macros.rs [hunk 2]
```diff
@@ -17,7 +17,16 @@ macro_rules! error {
...
+            let _ = ($($args),+);
```
[the remark this answers is not in the document]
Comment (lrubasze) [reply scrypto/src/macros.rs]: Good point! Done

File: scrypto/src/macros.rs (hunk 1 above)
Comment (lrubasze) [reply scrypto/src/macros.rs:29]: Yeah, that is pretty surprising :) Done

**Pull request merged**